express-zod-api/src/diagnostics.ts

```diff
@@ -1,41 +1,72 @@
-import { FlatObject } from "./common-helpers";
+import { FlatObject, getRoutePathParams } from "./common-helpers";
 import { contentTypes } from "./content-type";
 import { assertJsonCompatible } from "./deep-checks";
 import { AbstractEndpoint } from "./endpoint";
 import { ActualLogger } from "./logger-helpers";
 
 export class Diagnostics {
-  #verified = new WeakSet<AbstractEndpoint>();
+  #verifiedEndpoints = new WeakSet<AbstractEndpoint>();
   constructor(protected logger: ActualLogger) {}
 
-  public check(endpoint: AbstractEndpoint, ctx: FlatObject): void {
-    if (!this.#verified.has(endpoint)) {
-      if (endpoint.getRequestType() === "json") {
-        try {
-          assertJsonCompatible(endpoint.getSchema("input"), "in");
-        } catch (reason) {
-          this.logger.warn(
-            "The final input schema of the endpoint contains an unsupported JSON payload type.",
-            Object.assign(ctx, { reason }),
-          );
-        }
+  public checkJsonCompat(endpoint: AbstractEndpoint, ctx: FlatObject): void {
+    if (this.#verifiedEndpoints.has(endpoint)) return;
+    if (endpoint.getRequestType() === "json") {
+      try {
+        assertJsonCompatible(endpoint.getSchema("input"), "in");
+      } catch (reason) {
+        this.logger.warn(
+          "The final input schema of the endpoint contains an unsupported JSON payload type.",
+          Object.assign(ctx, { reason }),
+        );
       }
-      for (const variant of responseVariants) {
-        for (const { mimeTypes, schema } of endpoint.getResponses(variant)) {
-          if (mimeTypes?.includes(contentTypes.json)) {
-            try {
-              assertJsonCompatible(schema, "out");
-            } catch (reason) {
-              this.logger.warn(
-                `The final ${variant} response schema of the endpoint contains an unsupported JSON payload type.`,
-                Object.assign(ctx, { reason }),
-              );
-            }
+    }
+    for (const variant of responseVariants) {
+      for (const { mimeTypes, schema } of endpoint.getResponses(variant)) {
+        if (mimeTypes?.includes(contentTypes.json)) {
+          try {
+            assertJsonCompatible(schema, "out");
+          } catch (reason) {
+            this.logger.warn(
+              `The final ${variant} response schema of the endpoint contains an unsupported JSON payload type.`,
+              Object.assign(ctx, { reason }),
+            );
           }
         }
       }
-      this.#verified.add(endpoint);
+    }
+    this.#verifiedEndpoints.add(endpoint);
+  }
+
+  public checkPathParams(
+    path: string,
+    endpoint: AbstractEndpoint,
+    ctx: FlatObject,
+  ): void {
+    const params = getRoutePathParams(path);
+    for (const param of params) {
+      const sample = { [param]: "123" };
+      const result = endpoint.getSchema("input").safeParse(sample);
+      if (!result.success) {
+        const issue = result.error.issues.find(
```

Comment:
this does not work when param name differs from the input.
we should do opposite, probably using `extractObjectSchema()`